src/python/vdo-setup-dmtest/vdo-setup-dmtest.py

```diff
@@ -0,0 +1,512 @@
+#! /usr/bin/env python3
+"""
+  Installs and sets up a dmtest-python environment for running vdo tests.
+
+  Copyright (c) 2023 Red Hat
+"""
+
+from __future__ import print_function
+
+import argparse
+import git
+import logging
+import os
+import stat
+import subprocess
+import sys
+import yaml
+
+from source.Logger import *
+from source.Utils import *
+
+RUN_DIR = os.getcwd()
+
+class vdoDmtest(object):
+  # Initialize variables that will be used throughout the script
+  config           = {} # Dictionary of configuration definitions in config.yaml
+  dmtestHealthWarn = [] # List of messages that should be displayed after `dmtest health` is run
+  envVars          = [] # List of environment variables that need to be set/updated
+  repoKeys         = [] # List of repo keys defined in the configuration file
+  
+
+  def __init__(self, args):
+    # Parse input arguments
+    (options, extra) = self.parseArgs(args)
+    self.configFile  = options.configFile
+    self.debug       = options.debug
+    self.returnEnv   = options.returnEnv
+    self.logFile     = options.logFile
+    self.cleanLog    = options.numDays
+
+    # Initialize the logging object
+    self.log = logging.getLogger('vdo-setup-dmtest')
+    configureLogger(self.logFile, self.debug)
+    self.log.info("Starting vdo-setup-dmtest!")
+
+    if (self.cleanLog > 0):
+      cleanLogs(self.logFile, self.cleanLog)
+
+    # Check for sudo priviledges
+    if os.geteuid() != 0 and runCommandIgnoringErrors("sudo -v").returncode != 0:
+      msg = "This script must be run by a user with sudo priviledges"
+      raise PermissionError(msg)
+
+    # Get the configuration
+    self.getConfig(configFile = self.configFile)
+    self.log.info("Configuration file read and processed")
+
+    # Set workspace parameter if not defined in the configuration
+    if not self.config['WORKSPACE']:
+      self.config['WORKSPACE'] = os.path.join(RUN_DIR, 'repos')
+
+    # Find all repo keys in the config object and store them in the self.repoKeys list
+    self.getRepoKeys()
+    self.log.debug("Repo keys found: {0}".format(self.repoKeys))
+
+    # Verify the configuration device parameters
+    self.verifyDevices()
+
+    # Verify self.config['LINUX_SOURCE'] exists and is defined
+    self.verifyLinuxSource()
+
+  def parseArgs(self, args):
+    """
+    Parse any input arguments and apply defaults
+    """
+    parser = argparse.ArgumentParser(
+      prog='vdo-setup-dmtest',
+      description='Installs and sets up a dmtest-python environment for running vdo tests.',
+      epilog='For additional information see README.md within the module directory.')
```

Comment:
You're using double quotes everywhere else in this def.  Perhaps you can change these to doubles to be consistent with the rest of this def and the script?

Reply:
Fixed in latest push.